### DISCUSSION TITLE
SearXNG Temporary failure in name resolution with IP

### DISCUSSION THREAD

**Discussion by anony-source** [open-webui/open-webui#5885] [Q&A]
Hi, I would like to use SearXNG with Open WebUI (v0.3.30).
Installed them form official docker image, configured the json output for SearXNG, websearch at Open WebUI configured according to the documentations they are up to date.

SearXNG works locally, from the Open WebUI docker CLI I'm able to do a curl request to it and receive the answer (by using IP or DNS).

Based on the tests there is network connection and working name resolution between the two docker containers.

If I try to use the search function I receive a name resolution error even if I use IP address.

Can someone help me to figure out why am I getting name resolution error when I use IP instead of DNS?

In the logs I see the following:

> DEBUG [open_webui.apps.rag.search.searxng] searching http:// I removed the IP from the log:8080/search
> 
> DEBUG [open_webui.main] Commit session after request
> 
> ERROR [open_webui.apps.rag.main] [Errno -3] Temporary failure in name resolution
> 
> Traceback (most recent call last):
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 1030, in store_web_search
> 
>     loader = get_web_loader(urls)
> 
>              ^^^^^^^^^^^^^^^^^^^^
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 850, in get_web_loader
> 
>     if not validate_url(url):
> 
>            ^^^^^^^^^^^^^^^^^
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 879, in validate_url
> 
>     return all(validate_url(u) for u in url)
> 
>            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 879, in <genexpr>
> 
>     return all(validate_url(u) for u in url)
> 
>                ^^^^^^^^^^^^^^^
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 868, in validate_url
> 
>     ipv4_addresses, ipv6_addresses = resolve_hostname(parsed_url.hostname)
> 
>                                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/app/backend/open_webui/apps/rag/main.py", line 886, in resolve_hostname
> 
>     addr_info = socket.getaddrinfo(hostname, None)
> 
>                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
>   File "/usr/local/lib/python3.11/socket.py", line 974, in getaddrinfo
> 
>     for res in _socket.getaddrinfo(host, port, family, type, proto, flags):
> 
>                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 
> socket.gaierror: [Errno -3] Temporary failure in name resolution
>

**Comment by anony-source** [comment]
I was able to make it work by using the "Web Search" v0.1.12 -tool as a workaround solution.

**Comment by sinister1990** [comment]
I've got the same error 
`HTTPConnectionPool(host='searxng', port=8080): Max retries exceeded with url: /search?q=weather+in+Zagreb%2C+Croatia&format=json&pageno=1&safesearch=1&language=en-US&time_range=&categories=&theme=simple&image_proxy=0 (Caused by NameResolutionError("<urllib3.connection.HTTPConnection object at 0x7f24313af8d0>: Failed to resolve 'searxng' ([Errno -3] Temporary failure in name resolution)"))`

**Reply by sinister1990** [reply]
Also I've noticed that searxng:8080 is not working but localhost:8080 is opening in the browser normally.
